finality-aleph/src/justification/requester.rs

```diff
@@ -32,89 +35,114 @@ impl<B: BlockT> JustificationRequestStatus<B> {
         Self {
             block_number: None,
             block_hash: None,
-            tries: 0,
+            block_tries: 0,
+            parent: None,
+            n_childred: None,
+            children_tries: 0,
             report_threshold: REPORT_THRESHOLD,
         }
     }
 
-    fn save_block_number(&mut self, num: NumberFor<B>) {
-        if Some(num) == self.block_number {
-            self.tries += 1;
+    fn save_children(&mut self, hash: B::Hash, n_childred: usize) {
+        if self.parent == Some(hash) {
+            self.children_tries += 1;
```

Comment:
Why do you assume that the same parent has the same number of children on a consecutive attempt?

Reply:
Fixed.

Reply:
Oh gods, the `>=` here is so confusing. Do I understand correctly that this can _never_ become lower, so the `>` par is not strictly speaking necessary? If so I would change this to `==`, the fail case is acceptable then anyway.

Reply:
I would still count this as another try for requesting children of `self.parent`, even if `n_children` is different. Now we won't. So I'm still for removing `self.n_children >= n_children ` and adding  `self.n_children = n_children;` here. Not a strong opinion, though, so what timorl said is also fine with me.

Reply:
Ok, @maciejnems convinced me. Changed.